api/datastore/sql/sql.go

```diff
@@ -365,33 +370,57 @@ func (ds *sqlStore) RemoveApp(ctx context.Context, appName string) error {
 		}
 
 		deletes := []string{
-			`DELETE FROM logs WHERE app_name=?`,
-			`DELETE FROM calls WHERE app_name=?`,
-			`DELETE FROM routes WHERE app_name=?`,
+			`DELETE FROM logs WHERE app_id=?`,
+			`DELETE FROM calls WHERE app_id=?`,
+			`DELETE FROM routes WHERE app_id=?`,
 		}
-
 		for _, stmt := range deletes {
-			_, err := tx.ExecContext(ctx, tx.Rebind(stmt), appName)
+			_, err := tx.ExecContext(ctx, tx.Rebind(stmt), app.ID)
 			if err != nil {
 				return err
 			}
 		}
+
 		return nil
 	})
 }
 
-func (ds *sqlStore) GetApp(ctx context.Context, name string) (*models.App, error) {
-	query := ds.db.Rebind(`SELECT name, config, created_at, updated_at FROM apps WHERE name=?`)
-	row := ds.db.QueryRowxContext(ctx, query, name)
+func getAppTx(tx *sqlx.Tx, ctx context.Context, app string, isAppID bool, a *models.App) error {
+	q := ""
+	if !isAppID {
+		q = appNameSelector
```

Comment:
i don't think we should allow this

Reply:
this method is used to only internally, we can use this whenever we need app by name  (`GetApp`) or by ID (`GetAppByID`).

Reply:
it's brittle to leave this in, we can slip into security holes in the future with this in existence